packages/nvidia_nat_mcp/src/nat/plugins/mcp/auth/auth_provider.py

```
@@ -264,8 +270,67 @@ def __init__(self, config: MCPOAuth2ProviderConfig):
+            token_endpoint_auth_method=getattr(self.config, "token_endpoint_auth_method", None),
+            redirect_uri=str(self.config.redirect_uri) if self.config.redirect_uri else "",
+            scopes=self._effective_scopes() or [],
+            use_pkce=bool(self.config.use_pkce),
+        )
+
+        authenticated_context: AuthenticatedContext = await self._flow_handler.authenticate(
+            config, AuthFlowType.OAUTH2_AUTHORIZATION_CODE)
+
+        auth_header = authenticated_context.headers.get("Authorization", "")
+        if not auth_header.startswith("Bearer "):
+            raise RuntimeError("Invalid Authorization header")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)